[MPS] Enable bfloat16 testing

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #136987
* #137070

By even further reducing precisions of imprecise FP16 ops, introducing new BF16_LOW_PRECISION_OPS category and marking BF16 tests as xfail for `divfloor_rounding`, `floor_divide` and `remainder`. 
I guess the nature of low-precision results, is that MPSGraph, unlike the rest of the PyTorch does not do accumulation over fp32 for reduction operations

malfet committed:
[MPS] Enable bfloat16 testing

[ghstack-poisoned]

malfet committed:
Update on "[MPS] Enable bfloat16 testing"




[ghstack-poisoned]

malfet committed:
Update on "[MPS] Enable bfloat16 testing"




[ghstack-poisoned]

malfet committed:
Update on "[MPS] Enable bfloat16 testing"




[ghstack-poisoned]

malfet committed:
Update on "[MPS] Enable bfloat16 testing"


By even further reducing precisions of imprecise FP16 ops, introducing new BF16_LOW_PRECISION_OPS category and marking BF16 tests as xfail for `divfloor_rounding`, `floor_divide` and `remainder`


[ghstack-poisoned]

albanD reviewed (approved):
Any chance we can fix that behavior? This will most likely change the numerics for end users significantly.
@kulinseth  ?

malfet committed:
Update on "[MPS] Enable bfloat16 testing"


By even further reducing precisions of imprecise FP16 ops, introducing new BF16_LOW_PRECISION_OPS category and marking BF16 tests as xfail for `divfloor_rounding`, `floor_divide` and `remainder`. 
I guess the nature of low-precision results, is that MPSGraph, unlike the rest of the PyTorch does not do accumulation over fp32 for reduction operations


[ghstack-poisoned]
